### ISSUE TITLE
update camera aspect ratio in Plan View and Cross Section

### ISSUE THREAD

**Issue by Algorush** [3DStreet/3dstreet#635]
![image](https://github.com/3DStreet/3dstreet/assets/32197767/fb176913-90ed-422f-90d9-f753b9babea6)

**Comment by vincentfretin** [comment]
See my comments https://github.com/3DStreet/3dstreet-editor/issues/185#issuecomment-2149788570

**Comment by Algorush** [comment]
> See my comments [3DStreet/3dstreet-editor#185 (comment)](https://github.com/3DStreet/3dstreet-editor/issues/185#issuecomment-2149788570)

Thanks, I read the comments but didn't look in detail. Yes, the parameters of the Ortho camera are different from the Perspective camera. Perhaps this is causing the problem. I studied ortho and perspective cameras while working with 3dtiles-loader, I'll try to fix this

**Comment by vincentfretin** [comment]
Good luck. I spend about 2 hours trying without success.

**Comment by vincentfretin** [comment]
Oh I see that EditorControls is also messing around the left bottom right top property of the ortho camera
https://github.com/3DStreet/3dstreet/blob/5d6bacc9ca41fbae0b28cb940d30d852fba1a6c0/src/editor/lib/EditorControls.js#L120-L134
I missed that code when I tried to fix the issue.

**Comment by vincentfretin** [comment]
https://github.com/3DStreet/3dstreet/blob/5d6bacc9ca41fbae0b28cb940d30d852fba1a6c0/src/editor/lib/viewport.js#L212
probably just need to be executed on resize

**Comment by Algorush** [comment]
I made it work. I'll add a PR later. in window resize I first calculate `FrustumSize`, then add parameters for ortho camera. But also need to use different `frustumSize` for Top view and Plane View in inspector:
```
        if (inspector.camera.isOrthographicCamera) {
          // for plane view: inspector.camera.right - inspector.camera.left
          const frustumSize = inspector.camera.top - inspector.camera.bottom;
          inspector.camera.left = -frustumSize * aspect / 2;
          inspector.camera.right = frustumSize * aspect / 2;
          inspector.camera.top = frustumSize / 2;
          inspector.camera.bottom = -frustumSize / 2;
        }
```
@vincentfretin, maybe you know how can I get the current camera mode (plane/top view)? I can get it via camera rotation, but perhaps there is a parameter for this somewhere?

**Comment by Algorush** [comment]
I fixed it with `Cross Section View` too. 
But I found another issue with ortho camera. If zoom in on the camera in `Plan view` and then switch to `Cross Section View`, the camera angle takes on random values:
![after_zoom](https://github.com/3DStreet/3dstreet/assets/32197767/bbe584df-4c94-4979-a8b3-0df73dbe98c4)

**Comment by Algorush** [comment]
Added PR here in 3DStreet. I was unable to clone the aframe repository due to an error during the download, perhaps a problem with my unstable connection in a rural area

**Comment by vincentfretin** [comment]
I'll make changes to your PR.
Also for reference in three.js editor:
https://github.com/mrdoob/three.js/blob/1845f6cd0525b5c73b9da33c40e198c360af29f1/editor/js/Viewport.js#L153-L179

**Comment by vincentfretin** [comment]
PR #694 fixes the issue.

**Comment by kfarr** [comment]
closed by #694
